mds/entrypoint.sh

```diff
@@ -34,6 +47,27 @@ if [ ! -e /var/lib/ceph/mds/ceph-$MDS_NAME/keyring ]; then
       # Generate the new MDS key
       ceph auth get-or-create mds.$MDS_NAME mds 'allow' osd 'allow *' mon 'allow profile mds' > /var/lib/ceph/mds/ceph-${MDS_NAME}/keyring
    fi
+
+fi
+
+# Create the Ceph filesystem, if necessary
+if [ $CEPHFS_CREATE -gt 0 ]; then
+   FS_EXISTS=$(ceph fs ls | grep -c name:.${CEPHFS_NAME},)
+   if [ $FS_EXISTS -eq 0 ]; then
+      # Make sure the specified data pool exists
+      ceph osd pool stats ${CEPHFS_DATA_POOL} >/dev/null
+      if [ $? -ne 0 ]; then
+         ceph osd pool create ${CEPHFS_DATA_POOL} ${CEPHFS_DATA_POOL_PG}
+      fi
+
+      # Make sure the specified metadata pool exists
```

Comment:
It shouldn't be... `grep -c` should return the count of matches, so 0 means it does not exist, >0 means it _does_ exist.  Unless I'm cognitively broken; it is still morning, here.